src/audio/drc/drc_math_hifi3.c

```diff
@@ -62,35 +50,49 @@ static inline ae_f32 log10_fixed(ae_f32 x)
 	 * fpminimax(log10(x), 5, [|SG...|], [1/2;sqrt(2)/2], absolute);
 	 * max err ~= 6.088e-8
 	 */
-	int32_t e;
-	int32_t lshift;
 	ae_f32 exp; /* Q7.25 */
 	ae_f32 acc; /* Q6.26 */
 	ae_f32 tmp; /* Q6.26 */
+	ae_f64 tmp64;
+	int bit = 31 - AE_NSAZ32_L(x);
+	int32_t e = bit - 26;
```

Comment:
it's allowed to be negative, right?

Reply:
yes,
AE_SRAA32 Operation:
AE_SRAA32 d, d0, a0 [ae2_slot0, Inst, ae_minislot2]
Shift right or left arithmetic (sign-extending), element-wise, two 32-bit elements of AE_DR register d0 by AR register a0, with the result placed in d. For a positive shift amount, the
value is shifted to the right. In case of a negative shift amount, the value is shifted to the left.
d.L = (a0 ≥ 0) ? (d0.L >>s a0) : (d0.L << −a0);